File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.java
class Solution {
    public int findMaxK(int[] nums) {
        //1. sort the array
        Arrays.sort(nums);
        
        //2. loop through the entire array in backwards order
        for(int i = nums.length-1;i>=0;i--){
            //3. check if the negative version of it is there
            if (nums[i]>0){
                int index = Arrays.binarySearch(nums, nums[i]*-1);
                if (index >=0){
                    return nums[i];
                }
            }else{
                break;
            }
            
        }
        
        return -1;
    }
}

File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        #1. Sort the array 
        nums.sort()
        
        #2. Get the max each time - backwards for loop
        for i in range(len(nums)-1,-1,-1):
            if nums[i]>0:
                try:
                    index = nums.index(-1*nums[i])
                except ValueError:
                    index=-1
                if index!=-1:
                    return nums[i]
            else:
                break
        
        return -1